src/core/raster/qgsrasterfilewriter.cpp

```diff
@@ -1218,7 +1218,31 @@ QStringList QgsRasterFileWriter::supportedFormatExtensions( const RasterFormatOp
       continue;
 
     const QString matched = match.captured( 1 );
-    extensions << matched;
+    extensions.insert( matched );
   }
-  return extensions;
+
+  QStringList extensionList( extensions.constBegin(), extensions.constEnd() );
+
+  std::sort( extensionList.begin(), extensionList.end(), [options]( const QString & a, const QString & b ) -> bool
+  {
+    if ( options & SortRecommended )
+    {
+      if ( a == QLatin1String( "tif" ) )
+        return true;
+      else if ( b == QLatin1String( "tif" ) )
+        return false;
+      if ( a == QLatin1String( "tiff" ) )
+        return true;
+      else if ( b == QLatin1String( "tiff" ) )
+        return false;
+      if ( a == QLatin1String( "gpkg" ) )
+        return true;
+      else if ( b == QLatin1String( "gpkg" ) )
+        return false;
```

Comment:
I don't know if these are necessary...

Reply:
This is a way to make sure tif and gpkg are on top of the list (as they are super common)

Reply:
oops, sorry,  withdrawing my previous comment. I read to fast your comment. I thought you asked why they were necessary...